core/src/main/java/com/linecorp/armeria/common/util/PathParamExtractor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.util;
```

Comment:
I wonder this class could be useful for non-HTTP. If not, how about moving it to `com.linecorp.armeria.server.http`? If this is useful on the client side as well, we could place it at `com.linecorp.armeria.common.http`.